x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_details/extract_job_details.js

```diff
@@ -61,7 +61,9 @@ export function extractJobDetails(job) {
   if (job.calendars) {
     calendars.items = job.calendars.map((c) => [
       '',
-      <Link to={`/settings/calendars_list/edit_calendar/${c}?_g=()`}>{c}</Link>,
+      <EuiLink href={basePath.prepend(`/app/ml/settings/calendars_list/edit_calendar/${c}?_g=()`)}>
```

Comment:
rather than using `Link` or `EuiLink` here, is there a better way to create this link?
i'm asking because we don't need `basePath` anywhere else in this PR, so why here?

Reply:
I think moving forward we will have to use the url with basePath for all the href. The EuiLink work here without causing a full page reload because we are wrapping the ml application inside `RedirectAppLinks`. Alternatively, we can use it as a button and navigate the user to the new page but that means user will no longer be able to ctrl+click or right click to open in new tab.

`RedirectAppLinks` intercepts any <a> tag and override the onClick behavior to use `navigatetoUrl`, and therefore requires the `href` to have the full base path https://github.com/smith/kibana/blob/765a70f7cc2583d394b96015a6fd7f5ab44662ba/docs/developer/best-practices/navigation.asciidoc#L21


Also, in this particular component the `basePath` is needed so the link will work in the management section. If not the `href` would be like `abc/app/management/insightsAndAlerting/jobsListLink/settings/calendars_list/edit_calendar/nov4?_g=()` which is invalid. So that would mean we have to check if the table is being rendered in the `ml` plugin or the management section, and either prepend or not prepend the path for it to work properly.